.github/workflows/ci-build.yaml

```diff
@@ -189,7 +189,6 @@ jobs:
     needs: [ changed-files, argo-images ]
     if: ${{ needs.changed-files.outputs.e2e-tests == 'true' }}
     runs-on: ubuntu-latest
-    timeout-minutes: 30
```

Comment:
My understanding is that argoproj shares the compute resources amongst all running github actions.

I do agree with increasing the timeout, but I'd rather limit it to a number smaller than 6 hours, say 60 minutes. This would prevent bad tests which run on for much longer than expected from just wasting cycles that could then be given to other github actions that are waiting to happen.

Reply:
> My understanding is that argoproj shares the compute resources amongst all running github actions.

I don't think that's true, because this is a public repository. Private repositories are limited to 2,000 minutes under the free plan ([source](https://docs.github.com/en/billing/managing-billing-for-your-products/managing-billing-for-github-actions/about-billing-for-github-actions)), but this limit doesn't apply to public repositories.

> GitHub Actions usage is free for standard GitHub-hosted runners in public repositories, and for self-hosted runners. For private repositories, each GitHub account receives a certain amount of free minutes and storage for use with GitHub-hosted runners, depending on the account's plan. Any usage beyond the included amounts is controlled by spending limits.

Anyway, I pushed 67722f146d15736e0b54471efa49833bb4a25c6c to reduce the timeout to 60 minutes

Reply:
I was assuming the [maximum runners limit of 20](https://docs.github.com/en/actions/administering-github-actions/usage-limits-billing-and-administration#usage-limits) would kick in. When a lot of PRs are in flight at once then we do see actions not starting, so something is causing that. I would like to prevent rogue CI processes hogging the runners.

Thank you for the change.

Reply:
If we see 60 is genuinely too small we can bump it again.

Reply:
I've mentioned the GH concurrency limits in https://github.com/argoproj/argo-workflows/issues/10807#issuecomment-2094289752 and elsewhere and can confirm we do see some queuing when there are many PRs or when releases are being run.

This might be worth noting somewhere in the contributor docs as a point of reference